lib/paginated.rb

```diff
@@ -0,0 +1,44 @@
+module Paginated
+  def filter_and_paginate(sym)
+    original_method = :"#{sym}_no_pagination"
+    alias_method original_method, sym
+    define_method sym do |page, text, *args|
+      results = send(original_method, *args)
+      filtered = self.class.filter(results, text)
+      self.class.paginate(filtered, page, @config_options)
+    end
+  end
+
+  def filter(results, text)
+    if text.empty?
+      results
+    elsif
```

Comment:
Would an `else` be clearer?

Reply:
Good point, there is no condition here so should definitely be an `else`.